### DISCUSSION TITLE
Access Denied - Telegram IP Protection by NovaGram.ga

### DISCUSSION THREAD

**Discussion by ellasolo** [skrtdev/NovaGram#6] [General]
Hello!
After $Bot = new \skrtdev\NovaGram\Bot(TOKEN); i got message:
Access Denied - Telegram IP Protection by NovaGram.ga

How fix it?
Sorry for my English

**Comment by skrtdev** [comment]
Sorry I didn't see it. 

It is normal, Novagram restricts access to your webhook to telegram in order to prevent "fake updates" attack: no one will be able to send fake updates to bot. You just have to set Webhook on your php file and it will magically work. If you have some kind of reverse proxy which makes impossible to get the original IP address from the script, set disable_ip_check to true in bot's settings

**Reply by Idk23r** [reply]
I did both but i'm still getting the Access denied message.
im hosting on 000webhost

**Reply by skrtdev** [reply]
> I did both but i'm still getting the Access denied message.

You'll always get that message. What do you expect to see? If you have correctly set webhook, bot will be working

**Reply by hanifsetyananda** [reply]
how to set the webhook? can you link the documentation?

**Reply by Jobians** [reply]
**Set Webhook**
`https://api.telegram.org/bot{my_bot_token}/setWebhook?url={url_to_send_updates_to}`

**For instance:**

`https://api.telegram.org/bot123456:ABC-DEF1234ghIkl-zyx57W2v1u123ew11/setWebhook?url=https://www.example.com`

**Comment by Jobians** [comment]
Add `"mode" => Bot::NONE` in $Bot
